src/NServiceBus.Core.Tests/Pipeline/Outgoing/AttachSenderRelatedInfoOnMessageTests.cs

```diff
@@ -56,12 +58,41 @@ public async Task Should_not_override_nsb_version_headerAsync()
             Assert.AreEqual(nsbVersion, message.Headers[Headers.NServiceBusVersion]);
         }
 
+        [Test]
+        public async Task Should_set_the_deliver_At_headerAsync()
+        {
+            var message = await InvokeBehaviorAsync();
+
+            Assert.True(message.Headers.ContainsKey(Headers.DeliverAt));
+        }
+
+        [Test]
+        public async Task Should_not_override_deliver_at_headerAsync()
+        {
+            var delayTime = TimeSpan.FromSeconds(2);
+            var deliverAt = DateTimeOffsetHelper.ToWireFormattedString(DateTimeOffset.UtcNow.Add(delayTime));
+            var message = await InvokeBehaviorAsync(new Dictionary<string, string>
+            {
+                {Headers.DeliverAt, deliverAt}
+            });
+
+            Assert.True(message.Headers.ContainsKey(Headers.DeliverAt));
+            Assert.AreEqual(deliverAt, message.Headers[Headers.DeliverAt]);
+        }
+
         static async Task<OutgoingMessage> InvokeBehaviorAsync(Dictionary<string, string> headers = null, CancellationToken cancellationToken = default)
         {
             var message = new OutgoingMessage("id", headers ?? new Dictionary<string, string>(), null);
+            var stash = new ContextBag();
+            var dispatchProperties = new DispatchProperties
+            {
+                DelayDeliveryWith = new DelayDeliveryWith(TimeSpan.FromSeconds(2))
+            };
+
+            stash.Set(dispatchProperties);
```

Comment:
It doesn't seem right to do this for each of the tests in this class. Seems like we need one test that does NOT add a delayed delivery with and guarantee that it does not have a DeliverAt, and then another test that ensures that it is added when there is a timeout.

Reply:
Added DispatchProperties as a parameter for the InvokeBehaviorAsync test method.  Also, did a check for null on dispatchProperties, so it doesn't get added to the ContextBag for all the tests.